classes/submission/Representation.inc.php

```diff
@@ -85,6 +85,33 @@ function getSubmissionId() {
 	}
 
 	/**
+	 * Set submission revision.
+	 * @param $submissionRevision int
+	 */
+	function setSubmissionRevision($submissionRevision) {
+		$this->setData('submissionRevision', $submissionRevision);
+	}
+
+	/**
+	 * Get submission revision.
+	 * @return int
+	 */
+	function getSubmissionRevision() {
+		return $this->getData('submissionRevision');
+	}
+
+	/**
+	 * Get the current revision id for a galley from submission
+	 * @param $contextId int
+	 * @return int
+	 */
+	function getCurrentVersionId($contextId = null) {
```

Comment:
I solved it like this:

`SELECT MAX(submission_revision) FROM submission_settings ... WHERE submission_id = ?`